File: Array/linchuanxin/Algorithm/day28_0628_大于等于阈值的子数组数目/day28_leetcode1343.java
package day28;

public class day28_leetcode1343 {

    public int numOfSubarrays(int[] arr, int k, int threshold) {

        int standSum = k * threshold;
        int count = 0,sum = 0;

        for (int i = 0; i < k; i++) {
            sum += arr[i];
        }

        if(sum >= standSum)
            count = 1;

        for (int j = k; j < arr.length; j++){
            sum -= arr[j-k];
            sum += arr[j];
            if(sum >= standSum){
                count++;
            }
        }

        return count;

    }
}

File: Array/linchuanxin/Algorithm/day28_0628_大于等于阈值的子数组数目/day28_leetcode1343.kt
package day28



fun numOfSubarrays(arr: IntArray, k: Int, threshold: Int): Int {
    var standSum = k * threshold
    var sum = 0
    var count=0

    for (i in 0..k-1){
        sum += arr[i];
    }

    if (sum >= standSum)
        count = 1;

    for (j in k..arr.size-1){
        sum -= arr[j-k]
        sum += arr[j]
        if(sum >= standSum)
            count++
    }
    return count;
}


fun main() {
    var arr: IntArray = intArrayOf(1,1,1,1,1)
    print(numOfSubarrays(arr,1,0))
}